exchanges/binanceus/binanceus_wrapper.go

```diff
@@ -0,0 +1,975 @@
+// GetFundingHistory returns funding history, deposits and
+// withdrawals
+func (bi *Binanceus) GetFundingHistory(ctx context.Context) ([]exchange.FundHistory, error) {
+	// Not Implemented in the Binanceus endpoint
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetWithdrawalsHistory returns previous withdrawals data
+func (bi *Binanceus) GetWithdrawalsHistory(ctx context.Context, c currency.Code) (resp []exchange.WithdrawalHistory, err error) {
+	w, err := bi.WithdrawalHistory(ctx, c, "", time.Time{}, time.Time{}, 0, 10000)
+	if err != nil {
+		return nil, err
+	}
+	for i := range w {
+		tm, err := time.Parse(binanceUSAPITimeLayout, w[i].ApplyTime)
+		if err != nil {
+			return nil, err
+		}
+		resp = append(resp, exchange.WithdrawalHistory{
+			Status:          fmt.Sprint(w[i].Status),
+			TransferID:      w[i].ID,
+			Currency:        w[i].Coin,
+			Amount:          w[i].Amount,
+			Fee:             w[i].TransactionFee,
+			CryptoToAddress: w[i].Address,
+			CryptoTxID:      w[i].ID,
+			CryptoChain:     w[i].Network,
+			Timestamp:       tm,
+		})
+	}
+	return resp, nil
+}
+
+// GetRecentTrades returns the most recent trades for a currency and asset
+func (bi *Binanceus) GetRecentTrades(ctx context.Context, p currency.Pair, assetType asset.Item) ([]trade.Data, error) {
+	const limit = 1000
+	tradeData, err := bi.GetMostRecentTrades(ctx, RecentTradeRequestParams{p, limit})
+	if err != nil {
+		return nil, err
+	}
+	resp := make([]trade.Data, len(tradeData))
+	for i := range tradeData {
+		resp[i] = trade.Data{
+			TID:          fmt.Sprint(tradeData[i].ID),
+			Exchange:     bi.Name,
+			AssetType:    assetType,
+			CurrencyPair: p,
+			Price:        tradeData[i].Price,
+			Amount:       tradeData[i].Quantity,
+			Timestamp:    tradeData[i].Time,
+		}
+	}
+
+	if bi.IsSaveTradeDataEnabled() {
+		err := trade.AddTradesToBuffer(bi.Name, resp...)
+		if err != nil {
+			return nil, err
+		}
+	}
+	sort.Sort(trade.ByDate(resp))
+	return resp, nil
+}
+
+// GetHistoricTrades returns historic trade data within the timeframe provided
+func (bi *Binanceus) GetHistoricTrades(ctx context.Context, p currency.Pair,
+	assetType asset.Item, timestampStart, timestampEnd time.Time) ([]trade.Data, error) {
+	req := AggregatedTradeRequestParams{
+		Symbol:    p,
+		StartTime: uint64(timestampStart.UnixMilli()),
+		EndTime:   uint64(timestampEnd.UnixMilli()),
+	}
+	trades, err := bi.GetAggregateTrades(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+	result := make([]trade.Data, len(trades))
+	exName := bi.Name
+	for i := range trades {
+		t := trades[i].toTradeData(p, exName, assetType)
+		result[i] = *t
+	}
+	return result, nil
+}
+
+// SubmitOrder submits a new order
+func (bi *Binanceus) SubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	var submitOrderResponse order.SubmitResponse
+	if err := s.Validate(); err != nil {
+		return nil, err
+	}
+	var timeInForce RequestParamsTimeForceType
+	if s.AssetType == asset.Spot {
```

Comment:
If you invert this if, you can return earlier and not have the main piece of code indented